src/solvers/smt2/smt2_conv.cpp

```diff
@@ -96,6 +96,7 @@ smt2_convt::smt2_convt(
     break;
 
   case solvert::CVC4:
+    use_array_of_bool = true;
```

Comment:
:pick: The change of logic used and the switch on of `use_array_of_bool` appear to be two separate changes which should be in two separate commits. Is `use_array_of_bool` enough to fix the problem on its own? I think the error returned from the solver was to do with invalid sorts rather than unavailable theories.